### PR TITLE
chore: migrate info command to TS

### PR DESCRIPTION
**Pull request by FLGMwt**
Summary:
---------

@thymikee I cribbed your ignore comments from #683. If this is the right approach, I'll claim a few more.

Test Plan:
----------

Green CI

### CODE REVIEW

File: packages/cli/src/commands/info/info.ts [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
+// @ts-ignore - JS file
 import releaseChecker from '../../tools/releaseChecker';
 
 const info = async function getInfo(
-  argv: Array<string>,
-  ctx: ConfigT,
-  options: {},
+  _argv: Array<string>,
+  ctx: Config,
+  _options: {},
```
Comment (thymikee) [inline packages/cli/src/commands/info/info.ts]:
```suggestion
```
If it's unused, we can remove it :)

**Comment by thymikee** [comment]
Let's wait for this PR https://github.com/react-native-community/cli/pull/686 to be merged and then rebase

**Comment by thymikee** [comment]
Merged, you can now rebae

**Comment by FLGMwt** [comment]
I'll wait for #684 for `src/index.js` => `src/index.ts` to fix flow

**Comment by Esemesek** [comment]
@FLGMwt merged mentioned PR. You should be unblocked.

File: packages/cli/src/commands/info/info.ts (hunk 1 above)
Comment (FLGMwt) [reply packages/cli/src/commands/info/info.ts]: 👍 wasn't sure if we preferred the same API for all commands but I see now a lot of the commands skip it

**Comment by FLGMwt** [comment]
Any thoughts on the unrelated snapshot failure anyone? https://circleci.com/gh/react-native-community/cli/9083?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```
    -       "pbxprojPath": "<<REPLACED>>/node_modules/react-native-test/ios/HelloWorld.xcodeproj/project.pbxproj",
    +       "pbxprojPath": "<<REPLACED>>/node_modules/react-native-test/customLocation/customProject.xcodeproj/project.pbxproj",
```

**Comment by Esemesek** [comment]
Everything passes locally for me. We'll just have to retrigger CircleCI.

**Pull request merged**